packages/protonplus/protonplus.pacscript

```diff
@@ -0,0 +1,47 @@
+pkgname="protonplus"
+gives="protonplus"
+_app_id="com.vysp3r.ProtonPlus"
+pkgver="0.4.11"
+pkgdesc="A simple Wine and Proton-based compatiblity tools manager for GNOME"
+arch=('x86_64')
+url='https://github.com/Vysp3r/ProtonPlus'
+license=('GPL-3.0-or-later')
+depends=('libadwaita-1-0' 'libarchive-dev' 'libgee-0.8-dev' 'libgtk-4-dev' 'gettext')
+makedepends=('meson'
+  'valac'
+  'gettext'
+  'libadwaita-1-dev'
+  'libarchive-dev'
+  'libgee-0.8-dev'
+  'libgtk-4-dev'
+  'libjson-glib-dev'
+  'libsoup-3.0-dev'
+  'desktop-file-utils')
+checkdepends=('appstream-util')
+source=("${pkgname}-${pkgver}.tar.gz::${url}/archive/refs/tags/v${pkgver}.tar.gz")
+sha256sums=('223c8da373a265de28591ba9269440c5cf79fa4f9168c00b99503a65982d4df1')
+maintainer=("xdavius <xdavus@github.com>")
+
+incompatible=("debian:bookworm" "ubuntu:jammy")
+
+prepare() {
+  mkdir -p "ProtonPlus-${pkgver}"/build
+}
+
+build() {
+  cd "ProtonPlus-${pkgver}"
+  meson --prefix=/usr build
+  meson compile -C build
+}
+
+check() {
+  cd "ProtonPlus-${pkgver}"
+  meson test -C build --print-errorlogs || :
```

Comment:
Doesn't preventing failure by doing `|| :` bypass the entire point of checks? @oklopfer any input?

Reply:
Not necessarily, often packages have certain tests that are known to fail.

However, this package only has 3 tests, and 2 of them are succeeding. the 1 that is failing is because it needs `external_connection=true`.

Reply:
> Not necessarily, often packages have certain tests that are known to fail.
> 
> However, this package only has 3 tests, and 2 of them are succeeding. the 1 that is failing is because it needs `external_connection=true`. So it should probably drop the always-succeed and just fix that.

```
3/3 Validate appstream file FAIL            0.10s   exit status 1
>>> ASAN_OPTIONS=halt_on_error=1:abort_on_error=1:print_summary=1 MALLOC_PERTURB_=4 UBSAN_OPTIONS=halt_on_error=1:abort_on_error=1:print_summary=1:print_stacktrace=1 /usr/bin/appstream-util validate data/com.vysp3r.ProtonPlus.appdata.xml
――――――――――――――――――――――――――――――――――――― ✀  ―――――――――――――――――――――――――――――――――――――
stdout:
data/com.vysp3r.ProtonPlus.appdata.xml: FAILED:
? style-invalid         : <image> has vertical padding [https://raw.githubusercontent.com/Vysp3r/ProtonPlus/main/data/previews/Preview-1.png]
? style-invalid         : <image> has horizontal padding [https://raw.githubusercontent.com/Vysp3r/ProtonPlus/main/data/previews/Preview-1.png]
? style-invalid         : <ul> cannot start a description [(null)]
? style-invalid         : Not enough <p> tags for a good description [0/1]
stderr:
Validation of files failed
――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――


Summary of Failures:

3/3 Validate appstream file FAIL            0.10s   exit status 1

Ok:                 2   
Expected Fail:      0   
Fail:               1   
Unexpected Pass:    0   
Skipped:            0   
Timeout:            0   
```

Reply:
I don't know how to fix this test...

Reply:
Well it is certainly doing something different now that it has external connection, but yes it is still failing.

@Elsie19 this is why sometimes, “not necessarily”